src/Raven.Server/Documents/ETL/Providers/AI/GenAi/GenAiTask.cs

```diff
@@ -0,0 +1,392 @@
+        List<string> collections)
+    {
+        throw new NotSupportedException($"{nameof(ConvertAttachmentTombstonesEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<AiEtlItem> ConvertCountersEnumerator(DocumentsOperationContext context, IEnumerator<CounterGroupDetail> counters,
+        string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertCountersEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<AiEtlItem> ConvertTimeSeriesEnumerator(DocumentsOperationContext context, IEnumerator<TimeSeriesSegmentEntry> timeSeries,
+        string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertTimeSeriesEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override IEnumerator<AiEtlItem> ConvertTimeSeriesDeletedRangeEnumerator(DocumentsOperationContext context,
+        IEnumerator<TimeSeriesDeletedRangeItem> timeSeries, string collection)
+    {
+        throw new NotSupportedException($"{nameof(ConvertTimeSeriesDeletedRangeEnumerator)} is not supported for {nameof(EmbeddingsGenerationTask)}");
+    }
+
+    protected override EtlTransformer<AiEtlItem, GenAiScriptResult, GenAiStatsScope, GenAiPerformanceOperation>
+        GetTransformer(DocumentsOperationContext context)
+    {
+        return new GenAiScriptTransformer(Database, context, Transformation, null, Configuration);
+    }
+
+    protected override string LoadFailureMessage => $"Failed to generate embeddings in '{Configuration.Name}' task. Going to do the retry in {FallbackTime} (failure #{_fallbackCounter}).";
+
+    protected override void EnterFallbackMode(Exception e, DateTime? lastErrorTime)
+    {
+        // rate limits in embeddings are usually expressed as requests per minute or tokens per minute
+        // and they are reset on each full minute ticks, so we'll wait until the next full minute to retry
+        // at a minimum
+        int secondsToWaitToNextMinute = 60 - DateTime.UtcNow.Second;
+
+        var secondsToWait = ++_fallbackCounter switch
+        {
+            // first - we'll wait for the next minute each time - 5 minutes total 
+            < 5 => secondsToWaitToNextMinute,
+            // then - we'll wait for ~two minutes - 10 minutes total
+            < 10 => secondsToWaitToNextMinute + 60,
+            // then - we'll wait for three minutes - 30 minutes
+            < 20 => secondsToWaitToNextMinute + 120,
+            // finally - we'll use log2 minutes - so at 20+ failures, wait 5 minutes - 1 hour total
+            // then after 32 failures, wait 6 minutes each time - 3.2 hours, etc...
+            _ => secondsToWaitToNextMinute + ((int)Math.Log2(_fallbackCounter) * 60)
+        };
+
+        double max = Database.Configuration.Ai.EmbeddingsGenerationMaxFallbackTime.AsTimeSpan.TotalSeconds;
```

Comment:
Should it be a dedicated configuration option? Specific to Gen AI feature.

Reply:
removed this
there's no dedicated configuration option for now